app/src/main/java/com/example/final_project/AdminActivity.java

```diff
@@ -151,6 +185,91 @@ public void onClick(DialogInterface dialog, int which) {
         alertBuilder.create().show();
     }
 
+    private void showPromoteDialog(){
+        Button promoteButton = findViewById(R.id.promoteUserButton);
+        promoteButton.setOnClickListener(v -> {
+            if (selectedUsers.isEmpty()) {
+                Toast.makeText(this, "No users selected", Toast.LENGTH_SHORT).show();
+                return;
+            }
+            new AlertDialog.Builder(this)
+                        .setTitle("Confirm Promotion")
+                        .setMessage("Are you sure you want to promote this user?") //change to show selected user
+                        .setPositiveButton("Yes", (dialog, which) -> {
+                            for (User user : selectedUsers) {
+                                viewModel.promoteUser(user);
+                                }
+                            Toast.makeText(this,"User has been promoted", Toast.LENGTH_SHORT).show(); //change to show selected user
+                            selectedUsers.clear();
+                            })
+                        .setNegativeButton("Cancel", null).show();
+            });
+
+    }
```

Comment:
Just removing these listeners inside the method seems to fix this issue
```suggestion
    private void showPromoteDialog(){
            if (selectedUsers.isEmpty()) {
                Toast.makeText(this, "No users selected", Toast.LENGTH_SHORT).show();
                return;
            }
            new AlertDialog.Builder(this)
                        .setTitle("Confirm Promotion")
                        .setMessage("Are you sure you want to promote this user?") //change to show selected user
                        .setPositiveButton("Yes", (dialog, which) -> {
                            for (User user : selectedUsers) {
                                viewModel.promoteUser(user);
                                }
                            Toast.makeText(this,"User has been promoted", Toast.LENGTH_SHORT).show(); //change to show selected user
                            selectedUsers.clear();
                            })
                        .setNegativeButton("Cancel", null).show();
          

    }
```